File: Advanced_Prog_Assignment3-csharp/EmployeeManagementSystem/Models/AdminEmployee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using EmployeeManagementSystemCSharp.EmploymentTypes;
using EmployeeManagementSystemCSharp.Exceptions;

namespace EmployeeManagementSystemCSharp.Models
{
    public class AdminEmployee : Employee
    {
        private readonly List<string> _responsibilities = new List<string>();

        public AdminEmployee(string firstName, string lastName, DateTime dateOfBirth,
                             string employeeID, Position position, Manager manager,
                             EmploymentType emplType)
            : base(firstName, lastName, dateOfBirth, employeeID, position, manager, emplType)
        {
        }

        public void addResponsibility(string responsibility)
        {
            try
            {
                if (string.IsNullOrWhiteSpace(responsibility))
                {
                    throw new InvalidExceptions("Responsibility cannot be empty");
                }
                _responsibilities.Add(responsibility);
            }
            catch (InvalidExceptions e)
            {
                System.Console.Error.WriteLine("Error: " + e.Message);
            }
        }

        public List<string> getResponsibilities()
        {
            return new List<string>(_responsibilities);
        }

        public override void reportToManager(string message)
        {
            System.Console.WriteLine(
                $"Admin employee {getFirstName()} {getLastName()} reports to manager: {message}");
        }
    }
}

File: ASSIGNMENT3/EmployeeManagementSystem/src/AdminEmployee.java
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class AdminEmployee extends Employee {

    private List<String> responsibilities = new ArrayList<>();

    public AdminEmployee(String firstName, String lastName, Date dateOfBirth,
            String employeeID, Position position, Manager manager,
            EmploymentType emplType) {
        super(firstName, lastName, dateOfBirth, employeeID, position, manager, emplType);
    }

    public void addResponsibility(String responsibility) {
        if (responsibility != null && !responsibility.isEmpty()) {
            responsibilities.add(responsibility);
        }
    }

    public List<String> getResponsibilities() {
        return new ArrayList<>(responsibilities);
    }
}
